clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -659,10 +617,8 @@ private void handleFetchResponse(ClientResponse resp, FetchRequest request) {
                                         keyByteArray == null ? ConsumerRecord.NULL_SIZE : keyByteArray.length,
                                         valueByteArray == null ? ConsumerRecord.NULL_SIZE : valueByteArray.length,
                                         key, value);
-        } catch (KafkaException e) {
-            throw e;
         } catch (RuntimeException e) {
-            throw new KafkaException("Error deserializing key/value for partition " + partition + " at offset " + logEntry.offset(), e);
+            throw new SerializationException("Error deserializing key/value for partition " + partition + " at offset " + logEntry.offset(), e);
```

Comment:
In `logEntry.record().ensureValid()`, we throw `InvalidRecordException`. Do we really want to wrap that in a `SerializationException`?